src/linker/Linker.Steps/SweepStep.cs

```diff
@@ -449,15 +449,13 @@ static void UpdateTypeScope (TypeReference type, AssemblyDefinition assembly)
 			if (type.IsWindowsRuntimeProjection)
 				return;
 
-			if (type is GenericInstanceType git && git.HasGenericArguments) {
-				UpdateTypeScope (git.ElementType, assembly);
-				foreach (var ga in git.GenericArguments)
-					UpdateTypeScope (ga, assembly);
-				return;
-			}
+			if (type is TypeSpecification ts) {
+				UpdateTypeScope (ts.ElementType, assembly);
```

Comment:
This will crash for `FunctionPointerType`

Reply:
Well - I tried to write a test for that - but it crashes Roslyn first :wink: - https://github.com/dotnet/roslyn/issues/48765

Reply:
oh dear